src/config/config-helper.js

```diff
@@ -0,0 +1,158 @@
+import config from "../config/engine.json";
+import { create } from "../types/SortOption";
+
+/**
+ * This file abstracts most logic around the configuration of the Reference UI.
+ *
+ * Configuration is an important part of the "reusability" and "generic-ness" of
+ * the Reference UI, but if you are using this app as a starting point for own
+ * project, everything related to configuration can largely be thrown away. To
+ * that end, this file attempts to contain most of that logic to one place.
+ */
+
+export function getConfig() {
+  if (config.engineName) return config;
+
+  if (
+    typeof window !== "undefined" &&
+    window.appConfig &&
+    window.appConfig.engineName
+  ) {
+    return window.appConfig;
+  }
+}
+
+function toLowerCase(string) {
+  if (string) return string.toLowerCase();
+}
+
+function capitalizeFirstLetter(string) {
+  return string.charAt(0).toUpperCase() + string.slice(1);
+}
+
+export function getTitleField() {
+  // Use "title" field as the title field if a user hasn't specified one
+  return getConfig().titleField || "title";
+}
+
+export function getUrlField() {
+  return getConfig().urlField;
+}
+
+export function getUrlFieldTemplate() {
+  return getConfig().urlFieldTemplate;
+}
+
+export function getResultTitle(result) {
+  const titleField = getTitleField();
+
+  return (
+    result.getSnippet(titleField) ||
+    result.getRaw(titleField) ||
+    result.getRaw("id") // As a last resort, just show ID if nothing else
+  );
+}
+
+export function getResultUrl(result) {
+  const urlField = getUrlField();
+  if (urlField) return result.getRaw(urlField);
+
+  const urlFieldTemplate = getUrlFieldTemplate();
+
+  if (urlFieldTemplate) {
+    const fieldValueReplacementRegex = /{{([^}]*)}}/g;
+    let compiledUrlField = urlFieldTemplate;
+    let match = fieldValueReplacementRegex.exec(urlFieldTemplate);
+
+    while (match != null) {
+      compiledUrlField = compiledUrlField.replace(
+        match[0],
+        result.getRaw(match[1])
+      );
+      match = fieldValueReplacementRegex.exec(urlFieldTemplate);
+    }
+    return compiledUrlField;
+  }
+}
+
+// Because if a field is configured to display as a "title", we don't want
+// to display it again in the fields list
+export function stripUnnecessaryResultFields(resultFields) {
+  return Object.keys(resultFields).reduce((acc, n) => {
+    if (
+      [
+        "_meta",
+        "id",
+        toLowerCase(getTitleField()),
+        toLowerCase(getUrlField())
+      ].includes(toLowerCase(n))
+    )
+      return acc;
```

Comment:
Above on line 20, brackets were used when multiline like this. I think it looks more readable here too:

```
if (
  [
    "_meta",
    "id",
    toLowerCase(getTitleField()),
    toLowerCase(getUrlField())
  ].includes(toLowerCase(n))
) {
  return acc;
}
```